java/code/src/com/redhat/rhn/taskomatic/TaskoJob.java

```diff
@@ -106,8 +107,14 @@ private boolean isTaskRunning(TaskoTask task) {
     }
 
     private boolean isTaskThreadAvailable(TaskoTask task) {
-        return tasks.get(task.getName()) < Config.get().getInt("taskomatic." +
-                task.getTaskClass() + ".parallel_threads", 1);
+        int parallelThreads = Config.get().getInt("taskomatic." + task.getTaskClass() + ".parallel_threads", 1);
+        if (task.getClass().isAnnotationPresent(TaskConfigKey.class)) {
+            TaskConfigKey annotation = task.getClass().getAnnotation(TaskConfigKey.class);
```

Comment:
Is there any advantage to using an annotation rather then just a method returning the config name? With a method at least later on when every task has its config name and there is no default fallback method it would cause a compilation error not implementing it for any new task going forward.